datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -80,73 +78,6 @@ fn get_valid_types(
     signature: &TypeSignature,
     current_types: &[DataType],
 ) -> Result<Vec<Vec<DataType>>> {
-    fn array_append_or_prepend_valid_types(
```

Comment:
I see -- so one part of this PR is extracting this function into ArrayFunctionSignature ?